src/wallet/wallet.cpp

```diff
@@ -1155,6 +1155,30 @@ void CWallet::BlockConnected(const std::shared_ptr<const CBlock>& pblock, const
     for (size_t i = 0; i < pblock->vtx.size(); i++) {
         SyncTransaction(pblock->vtx[i], pindex, i);
     }
+
+
+    if (pindex) {
+        // Only notify UI if this transaction is in this wallet
+        if (hashPrevBestCoinbase.IsNull()) {
+            // For correctness we scan over the entire wallet, looking for
+            // the previous block's coinbase, just in case it is ours, so
+            // that we can notify the UI that it should now be displayed.
```

Comment:
> The commit message is correct, if its hard to read, any suggestions for improvements?

I still don't understand the commit message so I can't say how to improve it. But here is how it parses to me:

> Note that this does a full mapWallet loop after the first block
> which is connected after restart.

Fine so far.

> This is due to a previous bug where a coinbase transaction from the current best tip which existed in wallet immediately prior to restart would not be shown in the GUI

How is a previous bug relevant? Was it not fixed? What does "existed immediately prior to restart" mean? Does it mean a transaction _added_ immediately prior to restart? When would the transaction not be shown in the GUI, before or after the restart?

> except because on-load -checkblocks would call DisconnectBlock and ConnectBlock on the current tip.

Are there words missing between "except" and "because"? Is this saying the transaction would show up after the restart but not before?

> To avoid making it worse (ie that such transactions would never be displayed
until restart), a scan to find such transactions was added.

Is this referring to the scan you are adding in this commit, or some other scan?